modules/foreman_installation_medium.py

```diff
@@ -0,0 +1,194 @@
+options:
+  name:
+    description: The full installation medium name
+    required: true
+  locations:
+    description: List of locations the installation medium should be assigned to
+    required: false
+    default: None
+    type: list
+  operatingsystems:
+    description: List of operating systems the installation medium should be assigned to
+    required: false
+    default: None
+    type: list
+  organizations:
+    description: List of organizations the installation medium should be assigned to
+    required: false
+    default: None
+    type: list
+  path:
+    description: Path to the installation medium
+    required: false
+    default: None
+    type: string
+  server_url:
+    description: foreman url
+    required: true
+  username:
+    description: foreman username
+    required: true
+  password:
+    description: foreman user password
+    required: true
+  verify_ssl:
+    description: verify ssl connection when communicating with foreman
+    default: true
+    type: bool
+  state:
+    description: installation medium presence
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: create new debian medium
+  foreman_installation_medium:
+    name: "wheezy"
+    locations:
+      - "Munich"
+    organizations:
+      - "ATIX"
+    operatingsystems:
+      - "Debian"
+    path: "http://debian.org/mirror/"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    verify_ssl: False
+    state: present
+'''
+
+RETURN = ''' # '''
+
+try:
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        ping_server,
+        find_domain,
+        find_installation_medium,
+        find_locations,
+        find_organizations,
+        find_operating_systems,
+        naildown_entity_state,
+        sanitize_entity_dict,
+    )
+
+    from nailgun.entities import (
+        Media,
+    )
+
+    has_import_error = False
+except ImportError as e:
+    has_import_error = True
+    import_error_msg = str(e)
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'locations': 'location',
+    'name': 'name',
+    'organizations': 'organization',
+    'operatingsystems': 'operatingsystem',
+    'oses': 'os_family',  # the foreman community templates are using oses instead of os_family (which is wrong?)
+    'os_family': 'os_family',
+    'path': 'path_',
+}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+
+            name=dict(required=True),
+            locations=dict(type='list'),
+            organizations=dict(type='list'),
+            operatingsystems=dict(type='list'),
+            os_family=dict(),
```

Comment:
This is missing in the Documentation.

Reply:
Is this actually required? `foreman_provisioning_template` doesn't seem to need it.

Reply:
I'll try to derive `os_family` from `operatingsystems` though.